Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -1342,39 +1339,296 @@ namespace OpenInfraPlatform
+					carve::geom::vector<3> lastOffsetPoint = lastStartPoint - lastOffset;
+
+					offsetCurve3DPoints.push_back(lastOffsetPoint);
+					GeomUtils::appendPointsToCurve(offsetCurve3DPoints, targetVec);
+					segmentStartPoints.push_back(offsetCurve3DPoints[0]);
+				}
+
+#if defined(OIP_MODULE_EARLYBINDING_IFC4X3_RC4)
+				// IfcOffsetCurveByDistances SUBTYPE OF IfcOffsetCurve
+				/**********************************************************************************************/
+				/*! \brief Converts an \c IfcOffsetCurveByDistances to a tesselated curve.
+				* \param[in] offsetCurveByDistances	A pointer to data from \c IfcOffsetCurveByDistances.
+				* \param[out] targetVec				The tessellated line.
+				* \param[out] segmentStartPoints	The starting points of separate segments.
+				* \param[in] trim1Vec				The trimming of the curve as saved in IFC model - trim at start of curve.
+				* \param[in] trim2Vec				The trimming of the curve as saved in IFC model - trim at end of curve.
+				* \param[in] senseAgreement			Does the resulting geometry have the same sense agreement as the \c IfcOffsetCurveByDistances.
+				*
+				* \note The function is not implemented.
+				* \internal TODO.
+				*/
+				void convertIfcOffsetCurveByDistances(
+					const EXPRESSReference<typename IfcEntityTypesT::IfcOffsetCurveByDistances>& offsetCurveByDistances,
+					std::vector<carve::geom::vector<3>>& targetVec,
+					std::vector<carve::geom::vector<3>>& segmentStartPoints,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim1Vec,
+					const std::vector<std::shared_ptr<typename IfcEntityTypesT::IfcTrimmingSelect>>& trim2Vec,
+					const bool senseAgreement,
+					const typename IfcEntityTypesT::IfcTrimmingPreference& trimmingPreference
+				) const noexcept(false)
+				{
+					// **************************************************************************************************************************
+					//	https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/lexical/IfcOffsetCurveByDistances.htm
+					//	ENTITY IfcOffsetCurveByDistances
+					//		SUBTYPE OF(IfcOffsetCurve);
+					//			OffsetValues: LIST[1:? ] OF IfcPointByDistanceExpression;
+					//			Tag: OPTIONAL IfcLabel;
+					//	END_ENTITY;
+					// **************************************************************************************************************************
+
+					EXPRESSReference<typename IfcEntityTypesT::IfcCurve> basisCurve = offsetCurveByDistances->BasisCurve;
+					std::vector<carve::geom::vector<3>> basisCurvePoints;
+
+					convertIfcCurve(basisCurve, basisCurvePoints, segmentStartPoints,
+						trim1Vec, trim2Vec, senseAgreement, trimmingPreference);
+
+					std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcPointByDistanceExpression>> OffsetValues = offsetCurveByDistances->OffsetValues;
+					std::vector < std::vector<carve::geom::vector<3>>> offsetCurves;
+
+					for (auto OffsetValue : OffsetValues) {

```

Comment:
Only one OffsetCurve should be created 
You can start with considering only the case where OffsetValues has only one entry